xml/System.Threading/EventWaitHandleAcl.xml

```diff
@@ -41,14 +41,28 @@
         <Parameter Name="eventSecurity" Type="System.Security.AccessControl.EventWaitHandleSecurity" />
       </Parameters>
       <Docs>
-        <param name="initialState">To be added.</param>
-        <param name="mode">To be added.</param>
-        <param name="name">To be added.</param>
-        <param name="createdNew">To be added.</param>
-        <param name="eventSecurity">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="initialState">
+          <see langword="true" /> to set the initial state to signaled if the named event is created as a result of this call; <see langword="false" /> to set it to non-signaled.</param>
+        <param name="mode">One of the enum values that determines whether the event resets automatically or manually.</param>
+        <param name="name">The name, if the event is a system-wide synchronization event; otherwise, <see langword="null" /> or an empty string.</param>
+        <param name="createdNew">When this method returns, this argument is always set to <see langword="true" /> if a local event is created; that is, when <paramref name="name" /> is <see langword="null" /> or <see cref="F:System.String.Empty" />. If <paramref name="name" /> has a valid, non-empty value, this argument is set to <see langword="true" /> when the system event is created, or it is set to <see langword="false" /> if an existing system event is found with that name. This parameter is passed uninitialized.</param>
+        <param name="eventSecurity">The optional Windows access control security to apply.</param>
+        <summary>Gets or creates an <see cref="T:System.Threading.EventWaitHandle" /> instance, allowing a <see cref="T:System.Security.AccessControl.EventWaitHandleSecurity" /> instance to be optionally specified to set it during the event creation.</summary>
+        <returns>An object that represents a system event wait handle, if named, or a local event wait handle, if nameless.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+If a `name` is passed and the system event already exists, the existing event is returned. If `name` is `null` or <xref:System.String.Empty>, a new local event is always created.
+
+          ]]></format>
+        </remarks>
+        <exception cref="T:System.ArgumentNullException">.NET Framework only: The <paramref name="name" /> length is beyond MAX_PATH (260 characters).</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">The <paramref name="mode" /> enum value was out of legal range.</exception>
+        <exception cref="T:System.IO.DirectoryNotFoundException">Could not find a part of the path specified in <paramref name="name" />.</exception>
```

Comment:
I don't see this exception being raised by this API, the `name` parameter has nothing to do with directories on Windows, and on Unixes it's an implementation detail and probably should not result in such an exception

Reply:
When we reach here: https://github.com/dotnet/runtime/blob/8dc13298a89d50e64f24d0e755675bb01a33a399/src/libraries/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs#L82

We throw here: https://github.com/dotnet/runtime/blob/25d316cefafc4c3884f38acf7bdef971d2b9fe4b/src/libraries/Common/src/System/IO/Win32Marshal.cs#L39

Unit test verifying this exception added here: https://github.com/dotnet/runtime/blob/25d316cefafc4c3884f38acf7bdef971d2b9fe4b/src/libraries/System.Threading.AccessControl/tests/EventWaitHandleAclTests.cs#L70-L84

Reply:
I see thanks, I didn't know about that. If it can throw this exception for `EventWaitHandle` then it would throw in similar cases for the other types as well. This is fine for now, I'll look into fixing issues, filed issue https://github.com/dotnet/runtime/issues/35518.